### PR TITLE
generic preparations for EL8

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Review by cbarria — COMMENTED** [review]
most changes condition things to only happend on EL7 or 8, just that question about testing.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -144,6 +144,7 @@ def node_files
     include_examples 'krb5.conf content', %r{udp_preference_limit = 0}
   end
 
+  it { is_expected.to contain_class('epel') }
```
[the remark this answers is not in the document]
Comment (jhoblitt) [reply spec/spec_helper.rb:149]: That is a fair point.  We generally have been acting as if we only support `osfamily == Redhat` but it doesn't hurt to constraint new tests.

[the remark this answers is not in the document]
Comment (cbarria) [reply spec/spec_helper.rb:149]: oohh so, maybe we can add this on the future, but the rest of the PR looks good 👍

[the remark this answers is not in the document]
Comment (jhoblitt) [reply spec/spec_helper.rb]: I've reworked the `common` shared example to require a `facts` parameter and added negative tests to ensure that redhat-ism, at least from `profile::core::common` aren't present on non-EL OS'.

**Review by cbarria — APPROVED** [review]
lots of things but everything looks good 👍

**Pull request merged**